core/services/relay/evm/evm.go

```diff
@@ -977,6 +977,10 @@ func (r *Relayer) NewContractReader(ctx context.Context, chainReaderConfig []byt
 	return NewChainReaderService(ctx, r.lggr, r.chain.LogPoller(), r.chain.HeadTracker(), r.chain.Client(), *cfg)
 }
 
+func (r *Relayer) NewEVMChain(ctx context.Context) (commontypes.EVMChain, error) {
+	return NewEVMChain(ctx, r.lggr, r.chain.TxManager())
```

Comment:
Let's just return the Relayer here:
```suggestion
	return r, nil
```

Reply:
And perhaps this shouldn't be called `New` at all. What about something like `AsEVMChain`?